### PR TITLE
pss Change Color

### PR DESCRIPTION
**Pull request by CaptainKryga**
Added a change in the **color** of the liquid by the coefficient of the volume of the tank to the rate of entry.
Added **color** change accounting when adding several liquids at the same time.
Added a factor to speed up or slow down **color** changes.

### CODE REVIEW

**Comment by Macoron** [comment]
Oh boooyyyy

File: Assets/Unity Simple Liquid/Scripts/SplitController.cs [hunk 1]
```diff
@@ -269,9 +269,10 @@ private RaycastHit FindLiquidContainer(Vector3 splitPos, GameObject ignoreCollis
 					var liquid = hit.collider.GetComponent<SplitController>();
 					if (liquid && liquid != this)
 					{
-						
+						//color change in capacity
+						SendLiquidContainer(hit.collider.gameObject);
+
```
Comment (Macoron) [inline Assets/Unity Simple Liquid/Scripts/SplitController.cs]: This line can change color, even when liquids flow doesn't touch bottleneck. For example when bottle is upside down. I would recommend move it inside TransferLiquid function.
```
if (insideRadius)
{
	var lostAmount = liquidContainer.Volume * lostPercentAmount;
	liquid.liquidContainer.FillAmount += lostAmount;


	//color change in capacity
        SendLiquidContainer(hit.collider.gameObject);
}
```

File: Assets/Unity Simple Liquid/Scripts/SplitController.cs [hunk 2]
```diff
@@ -312,6 +313,17 @@ private RaycastHit FindLiquidContainer(Vector3 splitPos, GameObject ignoreCollis
 
 		#endregion
 
+		#region ChangeColor
+		private void SendLiquidContainer(GameObject go)
+		{
+			LiquidContainer lc = go.GetComponent<LiquidContainer>();
+			if (!lc)
+				return;
```
Comment (Macoron) [inline Assets/Unity Simple Liquid/Scripts/SplitController.cs]: It's better not GetComponent from go each frame, but pass LiquidContainer as parameter. Safe some performance.

File: Assets/Unity Simple Liquid/Scripts/LiquidContainer.cs [hunk 3]
```diff
@@ -338,6 +349,35 @@ private void OnDrawGizmosSelected()
...
+        #region ChangeColor
+        private List<LiquidContainer> listLC = new List<LiquidContainer>();
+        public List<LiquidContainer> GetColorList
+        {
+            get { return listLC; }
+        }
+
+        //playerMultiply for faster color change
+        [Range(0, 2)]
+        public float playerMultiply = 1;
```
Comment (Macoron) [inline Assets/Unity Simple Liquid/Scripts/LiquidContainer.cs]: Can you please rename this field and add tooltip attribute? From current name it's hard to tell what it do.

File: Assets/Unity Simple Liquid/Scripts/LiquidContainer.cs [hunk 4]
```diff
@@ -338,6 +349,35 @@ private void OnDrawGizmosSelected()
         }
         #endregion
 
+        #region ChangeColor
+        private List<LiquidContainer> listLC = new List<LiquidContainer>();
+        public List<LiquidContainer> GetColorList
+        {
+            get { return listLC; }
+        }
```
Comment (Macoron) [inline Assets/Unity Simple Liquid/Scripts/LiquidContainer.cs]: Why do you use list? Why not directly change color from SendLiquidContainer?

**Review by Macoron — CHANGES_REQUESTED** [review]
Hey @CaptainKryga, thank you for your contribution. It all works great, but I would like to request some minor changes.

File: Assets/Unity Simple Liquid/Scripts/SplitController.cs (hunk 1 above)
Comment (CaptainKryga) [reply Assets/Unity Simple Liquid/Scripts/SplitController.cs]: Good

File: Assets/Unity Simple Liquid/Scripts/LiquidContainer.cs (hunk 3 above)
Comment (CaptainKryga) [reply Assets/Unity Simple Liquid/Scripts/LiquidContainer.cs]: okay

File: Assets/Unity Simple Liquid/Scripts/SplitController.cs (hunk 2 above)
Comment (CaptainKryga) [reply Assets/Unity Simple Liquid/Scripts/SplitController.cs]: yes it turned out there is such a possibility and this is the best solution

File: Assets/Unity Simple Liquid/Scripts/LiquidContainer.cs (hunk 4 above)
Comment (CaptainKryga) [reply Assets/Unity Simple Liquid/Scripts/LiquidContainer.cs]: yes, I agree, it is wiser to do as you suggest, this is genius

**Comment by Macoron** [comment]
All good, merging!
Thank you @CaptainKryga! Great job!

**Pull request merged**